trustmanager/yubikey/yubikeystore.go

```diff
@@ -686,18 +711,32 @@ func (s *YubiKeyStore) addKey(keyID, role string, privKey data.PrivateKey) (
 	}
 	logrus.Debugf("Attempting to store key using yubikey slot %v", slot)
 
-	err = addECDSAKey(
-		ctx, session, privKey, slot, s.passRetriever, role)
-	if err == nil {
-		s.keys[privKey.ID()] = yubiSlot{
-			role:   role,
-			slotID: slot,
-		}
-		return true, nil
+	switch privKey.(type) {
+	case *data.ECDSAPrivateKey:
+		err = addECDSAKey(
+			ctx,
+			session,
+			privKey,
+			slot,
```

Comment:
The terminology is weird and we never settled on something. If memory serves (from discussions with the Yubikey folks 4-5 months back), in PKCS11 a "slot" is technically a device, i.e. the Yubikey itself is a PKCS11 "slot". I'm not sure what the correct term is for the space that a key actually resides in. We should find out and stop using "slot" because it'll just confuse anyone familiar with PKCS11.